### ISSUE TITLE
Exec freeze when 2 exec starts are executed in concurrence (with netty exec factory)

### ISSUE THREAD

**Issue by vuminhkh** [docker-java/docker-java#581]
Hello,

I'm running exec with this code
```java
ExecCreateCmdResponse execCreateCmdResponse = dockerClient.execCreateCmd(containerId)
                .withAttachStdin(true)
                .withAttachStdout(true)
                .withAttachStderr(true)
                .withCmd(command.split("\\s"))
                .exec();
        ExecStartCmd execStartCmd = dockerClient.execStartCmd(containerId)
                .withTty(true)
                .withExecId(execCreateCmdResponse.getId())
                .withDetach(false);
        if (input != null) {
            execStartCmd.withStdIn(input);
        }
        try (ResultCallbackTemplate resultCallback = execStartCmd.exec(new DockerStreamDecoder(log))) {
            resultCallback.awaitCompletion();
        } catch (InterruptedException e) {
            DockerExecutor.log.info("Command [{}] exec has been interrupted", command);
            throw new ArtifactInterruptedException("Script " + command + " exec has been interrupted", e);
        } catch (Exception e) {
            throw new ArtifactExecutionException("Script " + command + " exec encountered error", e);
        }
```

For the context, i'm running long installation scripts which perform download of binaries and then install them, so each command takes some minutes to finish.
If i'm running one command after another, everything is fine, but if i'm running 2 of them in concurrence then 
```java
resultCallback.awaitCompletion(); 
```
will never finish for one of the command.
Any idea of what happens ? Thank you for your helps.

Regards,

Minh Khang VU

**Comment by marcuslinke** [comment]
@vuminhkh Could you post some example code that shows how you run both in parallel please?

**Comment by vuminhkh** [comment]
@marcuslinke it's a little bit complicated for me to extract the code, I'll try to do it when I can. The thing is this bug is random and when it happens, I can't even remove the container with `docker rm -f`. I don't really know for the moment if it's a docker-java bug or docker bug.

**Comment by KostyaSha** [comment]
Check daemon logs.

**Comment by KostyaSha** [comment]
Usually it's devicemapper.

**Comment by vuminhkh** [comment]
I'm not seeing anything particular except it's saying that it failed to kill the container. Error at the end is when I tried to perform `docker exec -it test bash` after the `docker rm f` (which never returned).

`
May 16 20:46:04 ubuntu docker[20526]: time="2016-05-16T20:46:04.490888147Z" level=error msg="Handler for POST /containers/create returned error: Conflict. The name \"/test\" is already in use by container 0f0a333e5509410b08c762b6838c564324d955732ed05a3ed528948152c6626d. You have to remove (or rename) that container to be able to reuse that name."
May 16 20:46:28 ubuntu docker[20526]: time="2016-05-16T20:46:28.490763491Z" level=warning msg="container 0f0a333e5509410b08c762b6838c564324d955732ed05a3ed528948152c6626d restart canceled"
May 16 20:47:03 ubuntu docker[20526]: time="2016-05-16T20:47:03.210419991Z" level=warning msg="container ba8e2dcc63bd7b2909cf6cf17149502518551a7aa2e8d2024f529fbdbab4272f restart canceled"
May 16 20:47:32 ubuntu docker[20526]: time="2016-05-16T20:47:32.923420352Z" level=warning msg="container da64c38f5cb914128ceb10b1d5a2c117d62058f6ad78a3528a159d545ccfe607 restart canceled"
May 16 20:47:56 ubuntu docker[20526]: time="2016-05-16T20:47:56.050247886Z" level=warning msg="container 6f5bf5e64af1e31d0bba9a8b6ef50d7428b0c6927489901efb721ed65596241d restart canceled"
May 16 20:48:24 ubuntu docker[20526]: time="2016-05-16T20:48:24.380647839Z" level=warning msg="container 31dc31aaa819ac4f34ab7a533d705e0eb01689c5e8b8e37b8125cefb270fc213 restart canceled"
May 16 20:48:46 ubuntu docker[20526]: time="2016-05-16T20:48:46.474151860Z" level=warning msg="container 3e43af34fbd7002ff1930716c044ce02f6810e554a932420d8bed2a4b1c15233 restart canceled"
May 16 20:49:07 ubuntu docker[20526]: time="2016-05-16T20:49:07.765267126Z" level=warning msg="container 2aaee75aff18c11d6b9f3f73a081ef441471ce28a65cac6cf5e968e8bb914f2f restart canceled"
May 16 20:49:32 ubuntu docker[20526]: time="2016-05-16T20:49:32.211643569Z" level=warning msg="container 85b82b82ab04464ef05e5fc0c577635567e56b18b91a9e6d37a57d3fe6f84044 restart canceled"
May 16 20:58:37 ubuntu docker[20526]: time="2016-05-16T20:58:37.266909540Z" level=info msg="Container 68a4bcc5927a failed to exit within 10 seconds of kill - trying direct SIGKILL"
May 16 21:15:34 ubuntu docker[20526]: time="2016-05-16T21:15:34.013690971Z" level=warning msg="container deabfb30e61d6a7142f8eb3f8ff3cd396d13db5de45ff211c7fafcc9fe03d74e restart canceled"
May 16 21:46:23 ubuntu docker[20526]: time="2016-05-16T21:46:23.255132845Z" level=error msg="Error running exec in container: rpc error: code = 2 desc = \"oci runtime error: exec failed: exit status 1\"\n"
May 16 21:46:23 ubuntu docker[20526]: time="2016-05-16T21:46:23.255927893Z" level=error msg="Handler for POST /v1.22/exec/78b59fbb3b6c050793d19bdf463e8f72c795eb9f543bd0574df18858a5670538/start returned error: rpc error: code = 2 desc = \"oci runtime error: exec failed: exit status 1\""
May 16 21:46:23 ubuntu docker[20526]: 2016/05/16 21:46:23 http: response.WriteHeader on hijacked connection
May 16 21:46:23 ubuntu docker[20526]: 2016/05/16 21:46:23 http: response.Write on hijacked connection
May 16 21:46:23 ubuntu docker[20526]: time="2016-05-16T21:46:23.283846396Z" level=error msg="Handler for POST /v1.22/exec/78b59fbb3b6c050793d19bdf463e8f72c795eb9f543bd0574df18858a5670538/resize returned error: rpc error: code = 2 desc = \"containerd: processs not found for container\""
May 16 21:46:27 ubuntu docker[20526]: time="2016-05-16T21:46:27.769685179Z" level=error msg="Error running exec in container: rpc error: code = 2 desc = \"oci runtime error: exec failed: exit status 1\"\n"
May 16 21:46:27 ubuntu docker[20526]: time="2016-05-16T21:46:27.770259030Z" level=error msg="Handler for POST /v1.22/exec/0d792bda85e73e4fb8fc1f5303e455410e7c78ba6d653102023beb1c01575671/start returned error: rpc error: code = 2 desc = \"oci runtime error: exec failed: exit status 1\""
May 16 21:46:27 ubuntu docker[20526]: 2016/05/16 21:46:27 http: response.WriteHeader on hijacked connection
May 16 21:46:27 ubuntu docker[20526]: time="2016-05-16T21:46:27.771113425Z" level=error msg="Handler for POST /v1.22/exec/0d792bda85e73e4fb8fc1f5303e455410e7c78ba6d653102023beb1c01575671/resize returned error: rpc error: code = 2 desc = \"containerd: processs not found for container\""
May 16 21:46:27 ubuntu docker[20526]: 2016/05/16 21:46:27 http: response.Write on hijacked connection
`

**Comment by vuminhkh** [comment]
I'm not really sure but it seems related to https://github.com/docker/docker/issues/18180
When the problem happens, I saw on the machine zombie process + high cpu usage.
I upgraded the kernel version from 4.2.0-19-generic to 4.2.0-36-generic and I do not see the problem anymore. Sorry for the false alert. I'll continue to test to be sure that it does not happen anymore. Thank you very much for your help.
